database/gorm/query.go

```diff
@@ -701,6 +701,22 @@ func (r *QueryImpl) OrWhereIn(column string, values []any) ormcontract.Query {
 	return r.OrWhere(fmt.Sprintf("%s IN ?", column), values)
 }
 
+func (r *QueryImpl) WhereHas(table string, fk string, condition func(ormcontract.Query) ormcontract.Query) ormcontract.Query {
```

Comment:
S: We'd like to remove the `fk` field.

Reply:
I've found a solution to dynamically retrieve the foreign key field without needing to explicitly pass it. This approach involves passing both the parent model and the child model to the `WhereHas` method. Here's an example of how to use it:

```go
resultQuery, _ := query.WhereHas(User{}, Book{}, func(query ormcontract.Query) (ormcontract.Query, error) {
    modifiedQuery := query.Where("Name = ?", "Book A")
    return modifiedQuery, nil
})
 
```
With this modification, the foreign key field is dynamically determined based on the relationship between the parent and child models, eliminating the need for the `fk` parameter..

**i WOULD like to hear any feedback and any ideas how to improve further this method withou passing the Parent Model (User{})**